### PR TITLE
runtime+selfhost: Use `StringBuilder::appendff` instead of `append(format`

### PR DESCRIPTION
**Pull request by cg-jl**
This way we can remove a lot of duplicated effort where we format into a
String to then copy the string directly to the StringBuffer.

The parser had a little choke, where it would error in the rparen after
varargs: `..)`. It thought it had just passed over a comma, which was
the only possible case until now. `runtime/prelude.jakt` has `appendff`
which is the test for it passing.

### CODE REVIEW

**Comment by alimpfard** [comment]
🤔 you _can_ declare functions as variadic, why not add `extern fn appendff(mut this, anon format: StringView, ..) -> void`?

**Comment by cg-jl** [comment]
oh, I searched through the samples and didn't find any tests so I thought you had dropped them! my bad.

**Comment by alimpfard** [comment]
I'm pretty bad at adding tests for things I add "on the way" to a different feature...
oops!

File: selfhost/platform/posix_compiler.jakt [hunk 1]
```diff
@@ -1,84 +1,84 @@
-import jakt::path { Path }
```
[the remark this answers is not in the document]
Comment (cg-jl) [reply selfhost/platform/posix_compiler.jakt:1]: yeah, sorry. Couldn't track it down but must have been in between me debugging the errors and pulling your changes from yesterday

[the remark this answers is not in the document]
Comment (cg-jl) [reply selfhost/platform/posix_compiler.jakt:1]: All of the `platform/` ones are `sed -i 's/\r//g'`

**Comment by cg-jl** [comment]
Glad the test run under Windows. My build is broken under Windows; the linker shows garbage errors about `StaticDebug` and `StaticRelease` mixed with bad encoding:

<details><summary>sample</summary>

```
[ FAIL ] tests/codegen/type_pretty_printers.jakt
 : error LNK2038: se detectaron diferencias para 'RuntimeLibrary': el valor 'MTd_StaticDebug' no coincide con el valor 'MT_StaticRelease' en ns_level_circular_dependency-0b4664.o
LINK : warning LNK4098: la biblioteca predeterminada'libcmtd.lib' entra en conflicto con otras bibliotecas; use la biblioteca /NODEFAULTLIB:biblioteca
jakt_main_amd64-unknown-windows-unknown.lib(Main.cpp.obC:j\)U s:e resr\rgosru sL\NAKp2p0D3a8t:a \sLeo cdaelt\eTcetmapro\nj adkitfteersetn\cjiaakst tpeasrta- t'm_pI-T0E\RoAuTtOpRu_tD E:B UfGa_tLaElV EeLr'r:o re lL NvKa1l3o1r9 :'2 'd infoe rceonciinacsi ddee tceocnt aedla sv:a l3o4r
 '0' en optional_chaining_class-e4be19.o
jakt_main_amd64-unknown-windows-unknown.lib(Main.cpp.obj) : error LNK2038: se detectaron diferencias para 'RuntimeLibrary': el valor 'MTd_StaticDebug' no coincide con el valor 'MT_StaticRelease' en optional_chaining_class-e4be19.o
LINK : warning LNK4098: la biblioteca predeterminada'libcmtd.lib' entra en conflicto con otras bibliotecas; use la biblioteca /NODEFAULTLIB:biblioteca
jakt_main_amd64C-:u\nUksneorwsn\-gwsiunsd\oAwpsp-Duantkan\oLwonc.alli\bT(eMmapi\nj.ackptpt.eosbtj\)j a:k tetrersotr- tLmNpK-220\3o8u:tp uste  : dfeatteaclt aerrorno rd iLfNeKr1e3n1c9i:as  dpiafrear e'n_cIiTaEsR AdTeOtRe_cDtEaBdUaGs_:L E3V6E
L': el valor '2' no coincide con el valor '0' en optional_chaining-3b051e.o
jakt_main_amd64-unknown-windows-unknown.lib(Main.cpp.obj) : error LNK2038: se detectaron diferencias para 'RuntimeLibrary': el valor 'MTd_StaticDebug' no coincide con el valor 'MT_StaticRelease' en optional_chaining-3b051e.o
LINK : warning LNK4098: la biblioteca predeterminada'libcmtd.lib' entra en conflicto con otras bibliotecas; use la biblioteca /NODEFAULTLIB:biblioteca
```

</details>

The garbage is probably due to an encoding mishap since we use `SetConsoleOutputCp(UTF8)` and lld-link probably expects it to stay as `UTF16`.

**Pull request merged**